### DISCUSSION TITLE
Assign REALM user profile attribute to specific user

### DISCUSSION THREAD

**Discussion by klausveliu** [keycloak/keycloak#23681] [Q&A] [answered]
Hi,

Currently, I am using the declarative-user-profile feature of Keycloak.

In my implementation, I need to set these attributes for specific users (ideally for users assigned to a specific REALM group).

At the moment, it seems that when an attribute is created, it is being assigned to all users.

Is there any way to achieve this without developing any custom integration with Keycloak, or is it expected to be a feature in future releases of Keycloak?

Any hint is appreciated.

Thank you in advance!

**Comment by danielFesenmeyer** [comment, marked as answer]
It seems like it's currently only supported per scope - maybe that could work for you instead of a group?

I doubt whether it would make sense to have that dependent on a group, at least when registration is enabled, because new users won't have groups initially (at least per default).
